benchmarks/strings/bench_strings.sh

```diff
@@ -20,375 +20,263 @@ source ../bench_plot.sh
+function basic()
+{
+	for i in lib_variants/*; do
+		variant=`basename "$i" | cut -f 1 -d '.'`
+		echo "---------------------------------------------------------"
+		echo "  Trying variant $variant for benchmark $bench"
+		echo "---------------------------------------------------------"
+		git am $i
+		../../bin/nitc ../../examples/hello_world.nit
+		./hello_world
+		git reset --hard $head
+		rm hello_world
 	done
+}
 
-	prepare_res iter_ropes_iter.out iter_ropes_iter ropes_iter
-	if $verbose; then
-		echo "Ropes by iterator :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats (depth of the rope) = $i, Loops = $3"
-		fi
-		bench_command $i ropes_iter$i ./iteration_bench -m ropestr --iter-mode iterator --loops $2 --strlen $3 --ccts $i
-	done
+function prepare_compiler()
+{
+	cd ../../src
+	./ncall.sh
+	mv nitc.good ../bin/nitc
+	cd ../benchmarks/strings
+}
 
-	prepare_res iter_ropes_index.out iter_ropes_index ropes_index
-	if $verbose; then
-		echo "Ropes by index :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats (depth of the rope) = $i, Loops = $3"
-		fi
-		bench_command $i ropes_index$i ./iteration_bench -m ropestr --iter-mode index --loops $2 --strlen $3 --ccts $i
+function main()
+{
+	stop=false
+	while [ "$stop" = false ]; do
+		case "$1" in
+			-v) verbose=true; shift;;
+			-h) usage; exit;;
+			-n) count="$2"; shift; shift;;
+			*) stop=true
+		esac
 	done
 
-	prepare_res iter_buf_ropes_iter.out iter_buf_ropes_iter buf_ropes_iter
-	if $verbose; then
-		echo "RopeBuffer by iterator :"
+	if [ $# -lt 1 ]; then
+		usage;
+		exit;
 	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats (depth of the rope) = $i, Loops = $3"
-		fi
-		bench_command $i buf_ropes_iter$i ./iteration_bench -m ropebuf --iter-mode iterator --loops $2 --strlen $3 --ccts $i
-	done
 
-	prepare_res iter_buf_ropes_index.out iter_buf_ropes_index buf_ropes_index
-	if $verbose; then
-		echo "RopeBuffer by index :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats (depth of the rope) = $i, Loops = $3"
+	isok=false
+	for i in $benches; do
+		if [ $1 = $i ]; then
+			isok=true;
 		fi
-		bench_command $i buf_ropes_index$i ./iteration_bench -m ropebuf --iter-mode index --loops $2 --strlen $3 --ccts $i
 	done
-
-	plot iter.gnu
-
-	cd ..
-}
-
-function bench_substr()
-{
-	if [ -d string_substr ]; then
-		rm string_substr/*
-	else
-		mkdir string_substr
+	if [ "$isok" = false ]; then
+		usage;
+		exit;
 	fi
-	cd string_substr
 
-	if $verbose; then
-		echo "*** Benching substring performance ***"
-	fi
+	bench=$1
+	shift;
 
-	../../../bin/nitc --global ../substr_bench.nit
+	head=`git rev-parse HEAD`
 
-	prepare_res substr_flat.out substr_flat flatstring
-	if $verbose; then
-		echo "FlatStrings :"
+	if [ "$bench" = "basic" ]; then
+		basic;
```

Comment:
I think that the point of basic it to test that the compiler, the scripts, the patches and cie works, therefore I do not see the reason for a special case. If the point is to check that things are not broken, then one should check the real thing, not an simplified version (maybe rename `basic` as `check` also)